src/app/charts/line-graph.component.ts

```diff
@@ -42,121 +42,117 @@ export class LineGraphComponent {
   * and then we create a D3 Wrapper for our host element
   */
   constructor(private element: ElementRef) {
-    this.htmlElement = this.element.nativeElement;
-    this.host = D3.select(this.element.nativeElement);
+    self = this;
```

Comment:
This works, but very much not for the reason you expect. On this line, you don't declare it as `let self = this` which leads to the javascript default of `self` being declared as a global, which means that its accessible in this class, but also accessible _everywhere_. Which is not good, because it breaks encapsulation and clutters global scope. If you had declared it as `let self = this` then it would only be available in this function, and would be useless in all your other class methods.

Fortunately, there's a better way. And it's pretty nice in ES6. Anonymous arrow functions automatically bind to the `this` of the containing scope rather than the ES5 javascript default of binding to whatever the caller defines `this` to be, which defaults to `window`. So the somewhat hacky `var self = this` you would commonly see in older js code, such as:
```
function mapSomething() {
    var self = this;
    return _.map(this.data, function (d) { return self.someTransform(d); });
}
```
in ES6 can become the much cleaner:
```
function mapSomething() {
    return _.map(this.data, d => this.someTransform(d));
}
```

As always, the MDN docs are quite thorough, and worth read. More complete and detailed examples than the one provided above: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions